myapp/Classes/users.py

```diff
@@ -81,9 +81,17 @@ def filterUser(usertype):
     def searchUser(last_name):
         if last_name == "":
             return TypeError(
-                "you didn't select a usertype")
+                "You didn't write a last name!")
         # converts parameter to standard case and strips any whitespace before and after characters
         last_name_cleaned = last_name.title().strip()
+        users = User.objects.all()
+        counter = 0
+        for user in users:
```

Comment:
Is this check redundant? I would imagine the User.objects.filter() method returns a DoesNotExist exception of no user is found, therefore we do not need this check to throw the type error.

Other database API methods such as .get() throw a DoesNotExist exception of nothing is found with the given criteria, however I am less familiar with filter().